sklearn/tests/test_calibration.py

```diff
@@ -541,14 +542,18 @@ def test_calibration_pipeline(text_data, text_data_pipeline):
     Also test it can predict without running into validation errors.
     See https://github.com/scikit-learn/scikit-learn/issues/19637
     """
-    X, y = text_data
-    clf = text_data_pipeline
+    X, y = dict_data
+    clf = dict_data_pipeline
     calib_clf = CalibratedClassifierCV(clf, cv='prefit')
     calib_clf.fit(X, y)
     # Check attributes are obtained from fitted estimator
     assert_array_equal(calib_clf.classes_, clf.classes_)
-    msg = "'CalibratedClassifierCV' object has no attribute"
-    with pytest.raises(AttributeError, match=msg):
+
+    # Neither the pipeline nor the calibration meta-estimator
+    # expose the n_features_in_ check on this kind of data.
+    with pytest.raises(AttributeError):
+        clf.n_features_in_
```

Comment:
or `assert not hasattr(clf, 'n_features_in_')`? Not sure if the `pytest.raises(...)` makes it more clear.